devices/Sim7080/Sim7080G.cs

```diff
@@ -0,0 +1,301 @@
+        /// <param name="apiVersion">The Azure IoT Hub API version</param>
+        /// <param name="wait">The time to wait to establish the connection</param>
+        public void ConnectAzureIoTHub(string deviceId, string hubName, string sasToken, int portNumber = 8883, string apiVersion = "2021-04-12", int wait = 5000)
+        {
+            if (!_serialPort.IsOpen || NetworkConnected == ConnectionStatus.Disconnected)
+            {
+                EndpointConnected = ConnectionStatus.Disconnected;
+                return;
+            }
+
+            string username = $"{hubName}.azure-devices.net/{deviceId}/?api-version={apiVersion}";
+            string endpointUrl = $"{hubName}.azure-devices.net";
+
+            var retryCount = 0;
+
+            do
+            {
+                retryCount++;
+
+                EndpointConnected = SimController.EndpointConnect(_serialPort, deviceId, endpointUrl, portNumber, username, sasToken, wait);
+
+            } while (EndpointConnected == ConnectionStatus.Disconnected && retryCount < Retry);
+
+            //https://learn.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messages-c2d
+            SubTopic = $"devices/{deviceId}/messages/devicebound/#";
+
+            //https://learn.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messages-d2c
+            PubTopic = $"devices/{deviceId}/messages/events/";
+        }
+
+        /// <summary>
+        /// Connect to generic endpoint using MQTT protocol
+        /// </summary>
+        /// <param name="clientId">The device/client identifier</param>
+        /// <param name="endpointUrl">The endpoint URL</param>
+        /// <param name="portNumber">The MQTT port number</param>
+        /// <param name="username">The user name for endpoint authentication</param>
+        /// <param name="password">The password for endpoint authentication</param>
+        /// <param name="wait">The time to wait to establish the connection</param>
+        public void ConnectEndpoint(string clientId, string endpointUrl, int portNumber, string username, string password, int wait = 5000)
+        {
+            if (!_serialPort.IsOpen || NetworkConnected == ConnectionStatus.Disconnected)
+            {
+                EndpointConnected = ConnectionStatus.Disconnected;
+                return;
+            }
+
+            var retryCount = 0;
+
+            do
+            {
+                retryCount++;
+
+                EndpointConnected = SimController.EndpointConnect(_serialPort, clientId, endpointUrl, portNumber, username, password, wait);
+
+            } while (EndpointConnected == ConnectionStatus.Disconnected && retryCount < Retry);
+        }
+
+        /// <summary>
+        /// Subscribe to Cloud-2-Device Event Hub Topic
+        /// </summary>
+        /// <param name="topic">Event Hub Topic Name, uses <see cref="SubTopic"/></param> when null
+        public void Subscribe2Topic(string topic)
```

Comment:
as the implementation can be generic, all good to have this function, but if you add something for Azure, then please add the same API as in the Azure lib so it's going to be fully transparent.